src/Elastic.CommonSchema/Serialization/BaseJsonConverter.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Elastic.CommonSchema.Serialization
+{
+	internal partial class BaseJsonConverter : EcsJsonConverterBase<Base>
+	{
+		public override Base Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+		{
+			if (reader.TokenType == JsonTokenType.Null)
+			{
+				reader.Read();
+				return null;
+			}
+			if (reader.TokenType != JsonTokenType.StartObject) throw new JsonException();
+
+			var ecsEvent = new Base();
+
+			string loglevel = null;
+			DateTimeOffset? timestamp = default;
+			while (reader.Read())
+			{
+				if (reader.TokenType == JsonTokenType.EndObject)
+					break;
+
+				if (reader.TokenType != JsonTokenType.PropertyName)
+					throw new JsonException();
+
+				var read = ReadProperties(ref reader, ecsEvent, ref timestamp, ref loglevel);
```

Comment:
nit: `read` is ignored - not sure there could be anything done here if it's `false`.

Reply:
Aye still undecided what to here, we could ignore, throw, read unknown kv pairsand store on metadata. Will open a follow-up ticket.